test/Directory.Build.props

```diff
@@ -37,6 +37,14 @@
     <PackageReference Include="coverlet.msbuild" Version="$(CoverletVersion)" />
   </ItemGroup>
 
+  <ItemGroup>
+    <!-- Make sure to use the updated version of Microsoft.Extensions.* in test projects -->
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="$(MicrosoftExtensionsTestPackageVersion)" />
```

Comment:
Why does every test project have to reference these packages? Most of the test projects don't need these references.

Reply:
It didn't seem to hurt anything, and makes it easier to ensure that if the packages are needed, they use the correct version.

Reply:
> It didn't seem to hurt anything

That doesn't seem like a good reason.

Adding unnecessary dependencies causes people to start using them when they shouldn't be.

Reply:
If it's helpful for test code, does it matter?

Reply:
There’s only 1 project that needs these dependencies right? I think it would be different if there were multiple projects that needed these.

It also sets a pattern for the next person who needs to add a dependency to a project to just add it here and add it for all.

Reply:
It's hard for me to tell because of the transient nature of dependencies today. Otherwise I could just use this:

```xml
<PackageReference Update="Microsoft.Extensions.DependencyInjection.Abstractions" Version="$(MicrosoftExtensionsTestPackageVersion)" />
```

I'm not sure of a way to verify these are correct _without_ adding the dependencies to all test projects except by reviewing the full restore graph for each project.